server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -235,10 +235,15 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
                     "Auto-attach is disabled for owner \"{0}\"{1}.",
                     owner.getKey(), hypMessage));
             }
-            else {
+            else if (owner.isAutobindDisabled()) {
                 throw new AutobindDisabledForOwnerException(i18n.tr(
                     "Auto-attach is disabled for owner \"{0}\"{1}.",
-                    owner.getKey(), caMessage));
+                    owner.getKey()));
+            }
+            else {
+                log.debug("Auto-attach is disabled for owner \"{0}\"{1}.",
+                    owner.getKey(), caMessage);
+                return null;
```

Comment:
Looking at the code paths where this method is used in, returning null might be dangerous.
One example i can see is the `EntitleByProductsJob` job that could throw an NPE when it will try to call `.size()` on this list. An empty arraylist is safer.

Reply:
Done.